docs/dev/table/connectors/filesystem.md

```diff
@@ -0,0 +1,352 @@
+- The second is partition processing time. The time of partition commit according to partition
+creation time and current time. This trigger is more universal, but not so precise. For example,
+data delay or failover will lead to premature partition commit.
+
+#### Partition time extractor
+
+Time extractor defines extracting time from partition values.
+
+<table class="table table-bordered">
+  <thead>
+    <tr>
+        <th class="text-left" style="width: 20%">Key</th>
+        <th class="text-left" style="width: 15%">Default</th>
+        <th class="text-left" style="width: 10%">Type</th>
+        <th class="text-left" style="width: 55%">Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+        <td><h5>partition.time-extractor.kind</h5></td>
+        <td style="word-wrap: break-word;">default</td>
+        <td>String</td>
+        <td>Time extractor to extract time from partition values. Support default and custom. For default, can configure timestamp pattern. For custom, should configure extractor class.</td>
+    </tr>
+    <tr>
+        <td><h5>partition.time-extractor.class</h5></td>
+        <td style="word-wrap: break-word;"></td>
+        <td>String</td>
+        <td>The extractor class for implement PartitionTimeExtractor interface.</td>
+    </tr>
+    <tr>
+        <td><h5>partition.time-extractor.timestamp-pattern</h5></td>
+        <td style="word-wrap: break-word;"></td>
+        <td>String</td>
+        <td>The 'default' construction way allows users to use partition fields to get a legal timestamp pattern. Default support 'yyyy-mm-dd hh:mm:ss' from first field. If timestamp in partition is single field 'dt', can configure: '$dt'. If timestamp in partition is year, month, day, hour, can configure: '$year-$month-$day $hour:00:00'. If timestamp in partition is dt and hour, can configure: '$dt $hour:00:00'.</td>
+    </tr>
+  </tbody>
+</table>
+
+The default extractor provide a timestamp pattern way, you can construct timestamp pattern from your partition fields.
+
+You can also specify an implementation to complete partition time extraction. The custom extractor implementation like:
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+
+public class HourPartTimeExtractor implements PartitionTimeExtractor {
+	@Override
+    public LocalDateTime extract(List<String> keys, List<String> values) {
+        String dt = values.get(0);
+        String hour = values.get(1);
+		return Timestamp.valueOf(dt + " " + hour + ":00:00").toLocalDateTime();
+	}
+}
+
+{% endhighlight %}
+</div>
+</div>
+
+#### Partition commit policy
+
+The partiton commit policy defines how to commit a partition. Two policies are supported by default:
+
+- The first is metastore, but since the file system does not have metastore, it is an empty implementation.
+- The second is the success file, which will write an empty file in the directory corresponding to the partition.
```

Comment:
@JingsongLi @KurtYoung 

The naming of these options is not good. Metastore having a default no-op implementation does not make any sense. There should be `none`, `success-file`, and `custom`. I want to very strongly encourage making this change for 1.11.

Reply:
We can hide(throws exception?) it for filesystem connector.
But for Hive table, `metastore` is needed.

Reply:
Throwing exception in filesystem connector would be better, and we also need to highlight this in document.

Reply:
Created JIRA: https://issues.apache.org/jira/browse/FLINK-17925